Author: Karlis Peisenieks
Status: Working

This patch fixes a logic bug in NAT range calculations, which also
causes a large slowdown when ICMP floods go through NAT.
--- linux/net/ipv4/netfilter/ip_nat_core.c	26 Aug 2003 11:13:19 -0000	1.6
+++ linux/net/ipv4/netfilter/ip_nat_core.c	28 Aug 2003 08:25:45 -0000
@@ -159,8 +159,8 @@
 				continue;
 		}
 
-		if ((mr->range[i].flags & IP_NAT_RANGE_PROTO_SPECIFIED)
-		    && proto->in_range(&newtuple, IP_NAT_MANIP_SRC,
+		if (!(mr->range[i].flags & IP_NAT_RANGE_PROTO_SPECIFIED)
+		    || proto->in_range(&newtuple, IP_NAT_MANIP_SRC,
 				       &mr->range[i].min, &mr->range[i].max))
 			return 1;
 	}
